Autofill password generation support for iOS

amddg44 opened the pull request:
<!--
Note: This checklist is a reminder of our shared engineering expectations. Feel free to change it, although assigning a GitHub reviewer and the items in bold are required.

⚠️ If you're an external contributor, please file an issue first before working on a PR, as we can't guarantee that we will accept your changes if they haven't been discussed ahead of time. Thanks!
-->

Task/Issue URL: https://app.asana.com/0/72649045549333/1204452429715454/f
Tech Design URL:
CC:

**Description**:
Adds support for autofill password generation on iOS

<!--
Tagging instructions
If this PR isn't ready to be merged for whatever reason it should be marked with the `DO NOT MERGE` label (particularly if it's a draft)
If it's pending Product Review/PFR, please add the `Pending Product Review` label.

If at any point it isn't actively being worked on/ready for review/otherwise moving forward (besides the above PR/PFR exception) strongly consider closing it (or not opening it in the first place). If you decide not to close it, make sure it's labelled to make it clear the PRs state and comment with more information.
-->

**Steps to test this PR**:

Notes:
1. Modify `FeatureFlag.swift` and set `.autofillPasswordGeneration` to `.internalOnly`
2. Visit Browsing Menu -> Logins and delete logins between tests

_Use generated password and complete form_
1. Visit https://fill.dev/form/registration-username
2. Enter username
3. Tap on password field
4. Verify prompt appears offering a generated password
5. Accept to use it
6. Verify it populates the password and confirm password fields
7. Submit form
8. Verify toast shown confirming login saved
9. Tap on toast's `View` button; confirm you see the new login

_Use generated password but abandon form_
1. Visit https://fill.dev/form/registration-username
2. Enter username
3. Tap on password field
4. Verify prompt appears offering a generated password
5. Accept to use it
6. Verify it populates the password and confirm password fields
7. Don't submit form; instead, reload the page or navigate away
8. Visit  the Logins screen and verify the login with the username and generated password has been saved

_Use generated password then edit the generated password_
1. Visit https://fill.dev/form/registration-username
2. Enter username
3. Tap on password field
4. Verify prompt appears offering a generated password
5. Accept to use it
6. Manually edit the password and confirm password fields
7. Submit the form
8. Verify you are prompted if you want to save the login
9. Tap "not now" and verify no login was saved

_Decline to use generated password_
1. Visit https://fill.dev/form/registration-username
2. Enter username
3. Tap on password field
4. Verify prompt appears offering a generated password
5. Tap on "Not Now"; enter your own password in both fields
6. Submit form
7. Verify you are prompted if you want to save the login

_Copy generated password_
1. Visit https://fill.dev/form/registration-username
2. Tap on password field
3. Tap on the copy icon to the right of the password
4. Verify it copied successfully to the clipboard by pasting it somewhere

_Username updated later_
1. Visit https://fill.dev/form/registration-username
2. Don't enter username
3. Tap password and accept to use generated password
4. Then go enter a username and submit the form
5. Verify in Logins screen that the saved credential has both the username and password

_Confirm fireproof dialog does not overlay login saved toast_
1. In Settings > Fireproof > Fireproof Site toggle the `Ask to Fireproof sites on sign-in` to ON
2. Visit https://fill.dev/form/registration-username
3. Enter username
4. Tap on password field
5. Verify prompt appears offering a generated password
6. Accept to use it
7. Verify it populates the password and confirm password fields
8. Submit form
9. Verify toast shown confirming login saved 
10. Verify that prompt to fireproof displays after the toast has disappeared

_Disable toggle in autofill settings_
1. Go to the Logins screen and disable toggle for `Save and Autofill Logins`
2. Visit https://fill.dev/form/registration-username
3. Tap on password field; verify you are NOT prompted to use a generated password

_Smoke test existing save login flows_
1. Visit Logins screen and re-enable toggle for `Save and Autofill Logins`
2. Visit https://fill.dev/form/login-simple
3. Enter a username and password
4. Verify the prompt to save the login is presented
5. Save the login
6. Confirm the login has saved in the Logins screen

_Test remote config with config disabled_
1. Modify `FeatureFlag.swift` `.autofillPasswordGeneration` back to `.remoteReleasable(.subfeature(AutofillSubfeature.autofillPasswordGeneration))`
2. Visit https://fill.dev/form/registration-username
3. Tap on password field; verify you are NOT prompted to use a generated password

<!--
Before submitting a PR, please ensure you have tested the combinations you expect the reviewer to test, then delete configurations you *know* do not need explicit testing.

Using a simulator where a physical device is unavailable is acceptable.
-->

**Copy Testing**:

* [ ] Use of correct apostrophes in new copy, ie `’` rather than `'`

**Orientation Testing**:

* [ ] Portrait
* [ ] Landscape

**Device Testing**:

* [ ] iPhone SE (1st Gen)
* [ ] iPhone 8
* [ ] iPhone X
* [ ] iPhone 14 Pro
* [ ] iPad

**OS Testing**:

* [ ] iOS 14
* [ ] iOS 15
* [ ] iOS 16

**Theme Testing**:

* [ ] Light theme
* [ ] Dark theme

---
###### Internal references:
[Software Engineering Expectations](https://app.asana.com/0/59792373528535/199064865822552)
[Technical Design Template](https://app.asana.com/0/59792373528535/184709971311943)

amddg44 committed:
New password generation prompt screens + presentation logic

amddg44 committed:
New password generation strings

amddg44 committed:
Pixel events added

amddg44 committed:
Feature flag added

amddg44 committed:
Updated copy as per android ship review

amddg44 committed:
Merge remote-tracking branch 'origin/develop' into autofill/anya/password-generation

# Conflicts:
#	DuckDuckGo.xcodeproj/project.pbxproj
#	DuckDuckGo.xcodeproj/project.xcworkspace/xcshareddata/swiftpm/Package.resolved
#	DuckDuckGo/TabViewController.swift

amddg44 committed:
Updated for latest autofill js changes

amddg44 committed:
Layout improvements and new copy button

amddg44 committed:
Layout improvements and new copy button

amddg44 committed:
BSK bump

amddg44 committed:
Merge remote-tracking branch 'origin/develop' into autofill/anya/password-generation

# Conflicts:
#	DuckDuckGo.xcodeproj/project.pbxproj
#	DuckDuckGo.xcodeproj/project.xcworkspace/xcshareddata/swiftpm/Package.resolved

amddg44 committed:
Updated secureVaultManager delegate signatures

amddg44 committed:
BSK bump

amddg44 committed:
Delay fireproof prompt while toast is displayed

amddg44 committed:
Tidy up ordering of secureVaultDelegate functions

afterxleep reviewed (approved):
This is working as described.    Nicely done @amddg44 💪

amddg44 committed:
Merge remote-tracking branch 'origin/develop' into autofill/anya/password-generation

# Conflicts:
#	DuckDuckGo.xcodeproj/project.pbxproj
#	DuckDuckGo.xcodeproj/project.xcworkspace/xcshareddata/swiftpm/Package.resolved

amddg44 committed:
Merge remote-tracking branch 'origin/develop' into autofill/anya/password-generation

# Conflicts:
#	DuckDuckGo.xcodeproj/project.pbxproj
#	DuckDuckGo.xcodeproj/project.xcworkspace/xcshareddata/swiftpm/Package.resolved

amddg44 committed:
Translations for autofill password generation prompt (#1747)

Task/Issue URL: https://app.asana.com/0/0/1204613855344695/f
Tech Design URL:
CC:

Description:
Translations for the new autofill Password Generation prompt

amddg44 committed:
BSK set to 58.1.0

amddg44 committed:
Merge remote-tracking branch 'origin/develop' into autofill/anya/password-generation

# Conflicts:
#	DuckDuckGo.xcodeproj/project.pbxproj
#	DuckDuckGo.xcodeproj/project.xcworkspace/xcshareddata/swiftpm/Package.resolved

amddg44 committed:
BSK set to 58.1.0

amddg44 committed:
BSK set to 58.1.1
